tests/fuzz/fuzz_geo2.c

```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <string.h>
+
+#include "geos_c.h"
+
+static int initialized = 0;
+FILE * flogOut;
+
+void
+notice(const char *fmt, ...) {
+    va_list ap;
+    fprintf( flogOut, "NOTICE: ");
+    va_start (ap, fmt);
+    vfprintf( flogOut, fmt, ap);
+    va_end(ap);
+    fprintf( flogOut, "\n" );
+}
+
+void
+log_and_exit(const char *fmt, ...) {
+    va_list ap;
+    fprintf( flogOut, "ERROR: ");
+    va_start (ap, fmt);
+    vfprintf( flogOut, fmt, ap);
+    va_end(ap);
+    fprintf( flogOut, "\n" );
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
```

Comment:
I'm not sure I understand this test.
* Get a random hunk of binary data
* Try to parse it as WKT?
* If success... try to parse it as WKB?
* If success (how?) ... try to intersect/difference/union it with itself (bearing in mind those functions require valid inputs)
I guess I can see throwing garbage geometry at intersection/difference/union on the theory that they shouldn't crash ever, even if they might return nonsense. What I don't see is how this attempt to use the input data as both WKB and WKT works.

Reply:
> Try to parse it as WKT?
If success... try to parse it as WKB?

This is not the same data.

cf `GEOSGeomFromWKT(Data);` vs  `GEOSGeomFromWKB_buf(Data+sep, Size-sep);` where sep is the index of the first null character

Is this what you were missing ?

Reply:
Yes, kind of. But also, what is generating the inputs? Is there, like, a "fuzzy geometry generator"? If they are independent random inputs the odds of them ever being both valid at once seem very low. I guess I'm missing something important here.

Reply:
> Yes, kind of. But also, what is generating the inputs?

They are called the fuzzing engines.
They are libfuzzer, afl, honggfuzz.

> Is there, like, a "fuzzy geometry generator"? If they are independent random inputs the odds of them ever being both valid at once seem very low. I guess I'm missing something important here.

They are not independent random inputs.
The big secret is the coverage as a feedback.
You can search for tutorials about it

Reply:
For the sake of sanity when trying to debug the oss-fuzz result, please keep both inputs as WKT. 
At the end of the day, oss-fuzz will find something and a human will have to look at what's going on there.

It's ok to have more fuzzing targets for different kinds of inputs, just like PostGIS does.

Reply:
> please keep both inputs as WKT.

done